pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,399 @@
+
+	Operations []string
+	NodeNames  []string
+
+	// Supported operations
+	Schedulable bool
+	Evacuate    bool
+	ListPods    bool
+
+	// Optional arguments
+	Selector    util.StringFlag
+	PodSelector util.StringFlag
+	DryRun      bool
+}
+
+const manageNodeLongDesc = `Manage node operations: schedulable, evacuate and list-pods.
+
+Examples:
+
+	# Block accepting any new pods on nodes
+	$ %[1]s <mynode> --schedulable=false
+
+	# Move selected pods to a different node
+	$ %[1]s <mynode> --evacuate --pod-selector="<service=myapp>"
+
+	# List all pods on nodes
+	$ %[1]s <mynode1> <mynode2> --list-pods
+
+	# Mark selected nodes as schedulable
+	$ %[1]s --selector="<env=dev>" --schedulable=true
+`
+
+func NewCommandManageNode(f *clientcmd.Factory, commandName, fullName string, out io.Writer) *cobra.Command {
+	cfg := &NodeConfig{}
+
+	cmd := &cobra.Command{
+		Use:   commandName,
+		Short: "Manage node operations: schedulable, evacuate, list-pods",
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+
+			defaultNamespace, err := f.DefaultNamespace()
+			kcmdutil.CheckErr(err)
+
+			_, kc, err := f.Clients()
+			kcmdutil.CheckErr(err)
+
+			mapper, typer := f.Object()
+
+			cfg.DefaultNamespace = defaultNamespace
+			cfg.Kclient = kc
+			cfg.Writer = out
+			cfg.Mapper = mapper
+			cfg.Typer = typer
+			cfg.RESTClientFactory = f.Factory.RESTClient
+			cfg.Printer = f.Printer
+			cfg.Operations = []string{}
+			cfg.NodeNames = []string{}
+
+			if c.Flag("schedulable").Changed {
+				cfg.Operations = append(cfg.Operations, "schedulable")
+			}
+			if c.Flag("evacuate").Changed {
+				cfg.Operations = append(cfg.Operations, "evacuate")
+			}
+			if c.Flag("list-pods").Changed {
+				cfg.Operations = append(cfg.Operations, "list-pods")
+			}
+
+			if len(args) != 0 {
+				cfg.NodeNames = append(cfg.NodeNames, args...)
+			}
+
+			errList := []error{}
+			if errList = cfg.ValidateParams(); len(errList) != 0 {
+				err := fmt.Errorf("%v", errList)
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			err = cfg.RunManageNode()
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", false, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", false, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&cfg.ListPods, "list-pods", false, "List all/selected pods on the node.")
+
+	flags.BoolVar(&cfg.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
+	flags.Var(&cfg.PodSelector, "pod-selector", "Label selector to filter pods on the node. Optional parameters for --evacuate or --list-pods")
+	flags.Var(&cfg.Selector, "selector", "Label selector to filter nodes. Either pass one/more nodes as arguments or use this node selector")
+
+	return cmd
+}
+
+func (cfg *NodeConfig) ValidateParams() []error {
+	errList := []error{}
+	if cfg.Selector.Provided() {
```

Comment:
If we use len(selector)>0 instead of Provided(),  when user skips node names and selectors, we will apply default selector=""
openshift admin mange-node --schedulable=false   will mark all nodes as unschedulable. Are we okay with this behavior?